networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,139 @@
+    """
+    # Assert that the graph G is a tree
+    if not nx.is_tree(G):
+        NetworkXError("Your graph is not a tree")
+    # step 0
+    if G.number_of_nodes() == 2:
+        return 1, set(G.nodes())
+    elif G.number_of_nodes() == 1:
+        return 0, set(G.nodes())
+
+    # step 1
+    U = {node for node, deg in G.degree if deg == 1}
+    values = {n: 0 for n in U}
+    T = G.copy()
+    T.remove_nodes_from(U)
+
+    # step 2
+    W = {node for node, deg in T.degree if deg == 1}
+    values.update((w, G.degree[w] - 1) for w in W)
+
+    # step 3
+    while T.number_of_nodes() >= 2:
+        # step 4
+        w = min(W, key=lambda n: values[n])
+        v = next(T.neighbors(w))
+
+        # step 5
+        U.add(w)
+        W.remove(w)
+        T.remove_node(w)
+
+        # step 6
+        if T.degree(v) == 1:
+            # update t(v)
+            values.update({v: _get_max_broadcast_value(G, U, v, values)})
+            W.add(v)
+
+    # step 7
+    v = nx.utils.arbitrary_element(T)
+    b_T = _get_max_broadcast_value(G, U, v, values)
+    return b_T, _get_broadcast_centers(G, v, values, b_T)
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def tree_broadcast_time(G, node=None):
+    """Return the Broadcast Time of the tree G.
+
+    Broadcasting is an information dissemination problem in which one vertex in a graph, called the originator,
+    must distribute a message to all other vertices by placing a series of calls along the edges of the graph.
+    Once informed, other vertices aid the originator in distributing the message.
+    The broadcasting must be completed as quickly as possible subject to the following constraints:
+    - Each call requires one unit of time.
+    - A vertex can only participate in one call per unit of time.
+    - Each call only involves two adjacent vertices: a sender and a receiver.
+    The minimum broadcast time of a vertex is defined as the minimum amount of time required to complete
+    broadcasting starting from the originator.
+    The broadcast time of a graph is defined as the maximum broadcast time from all its vertices.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+        The graph should be an undirected tree
+    node: int
+        index of starting vertex. If none,
+        the algorithm returns the broadcast
+        time of the tree.
+    Returns
+    -------
+    BT : int
+        Broadcast Time of a vertex in a tree
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        If the graph is directed or is a multigraph.
+
+    References
+    ----------
+    .. [1] Slater, P.J., Cockayne, E.J., Hedetniemi, S.T,
+       Information dissemination in trees. SIAM J.Comput. 10(4), 692–701 (1981)
+    """
+    b_T, b_C = tree_broadcast_center(G)
+    if node is not None:
+        return b_T + min(nx.shortest_path_length(G, node, u) for u in b_C)
```

Comment:
Here I implement the algorithm for broadcast from a specific vertex. The intuition behind why this works is that the goal is to reach any vertex in the broadcast centre as fast as possible. (Hence the min distance from node to all the centres)

Reply:
Yea I think you are right about the intuition, if you were to construct a longest path to the broadcast centre like you mentioned, then maybe the broadcast centre won't be at the same place.. and it would fall somewhere in the middle of that path (to the hypothetical broadcast centre). 
Here is a proof from the paper that explains it in more detail, just for verification purposes.
<img width="1052" alt="image" src="https://github.com/networkx/networkx/assets/89562186/aaf064c1-8670-402c-9b76-3d3b604603f5">